distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
 
     }
 
+    @Test
+    public void testDownloadFromFederatedCswSource() throws Exception {
+        RestitoRetrievalResponse restitoRetrievalResponse =
+                new RestitoRetrievalResponse("testData");
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"))
+                .then(getCswRetrievalHeaders(), restitoRetrievalResponse.action());
+
+        String restUrl =
+                REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + CSW_STUB_RESOURCE_ID
+                        + "?transform=resource";
+
+        // Verify that the testData from the csw stub server is returned.
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is("testData"));
+        // @formatter:on
+    }
+
+    @Test
+    public void testSlowResponse() throws Exception {
+        RestitoRetrievalResponse slowAction = new RestitoRetrievalResponse("testData", 2000);
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"))
+                .then(getCswRetrievalHeaders(), slowAction.action());
+
+        String restUrl =
+                REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + CSW_STUB_RESOURCE_ID
+                        + "?transform=resource";
+
+        // Verify that the testData from the csw stub server is returned.
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is("testData"));
+        // @formatter:on
```

Comment:
Sorry, github is acting up for me. It's been attaching my comments to the wrong threads and this comment thread is not even inside to the `testSlowResponse()` method you're referring to.

I'm not sure why the 2000 ms test does not cause a failure. The retry logic inside the downloader seems to infer that it will time out if it's longer than 1 second between receiving data. If I set the response to something high like 10000ms, it does fail. I'll look into it a bit.